linkerd/stack/metrics/src/service.rs

```diff
@@ -35,9 +34,15 @@ where
         match self.inner.poll_ready(cx) {
             Poll::Pending => {
                 self.metrics.not_ready_total.incr();
-                if self.blocked_since.is_none() {
-                    self.blocked_since = Some(Instant::now());
+                // If the service was already pending, then add the time we
+                // waited and reset blocked_since. This allows the value to be
+                // updated even when we're "stuck" in pending.
+                let now = Instant::now();
+                if let Some(t0) = self.blocked_since.take() {
+                    let not_ready = now - t0;
+                    self.metrics.poll_millis.add(not_ready.as_millis() as u64);
```

Comment:
nit: should we check this cast? i guess if a stack has been blocked for > 18446744073709551615 milliseconds, that's probably pretty bad...